### PR TITLE
Update free-courses-en.md

### PR DESCRIPTION
**Pull request by MeenakshyBS**
added 3 courses that wasn't in the list

## What does this PR do?
Added 3 resources | Improved repo

### Why is this valuable (or not)?
It definitely helps programmers to self learn and practice
### How do we know it's really free?
Check the website it is free and authentic
## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -582,6 +583,7 @@
 * [Learn HTML and CSS](https://www.bitdegree.org/courses/coding-for-beginners-space-doggos) (BitDegree)
 * [Learn HTML5 Programming From Scratch](https://www.udemy.com/learn-html5-programming-from-scratch/)
 * [Learn to style HTML using CSS](https://developer.mozilla.org/en-US/docs/Learn/CSS)
+* [Responsive Web Design](https://www.freecodecamp.org/learn/2022/responsive-web-design) - FreeCodeCamp
```
Comment (eshellman) [inline courses/free-courses-en.md:586]: duplicate

File: courses/free-courses-en.md [hunk 2]
```diff
@@ -137,6 +137,7 @@
 * [IIT Bombay Foundation of Data Structures (CS213.1x)](https://courses.edx.org/courses/course-v1:IITBombayX+CS213.1x+1T2017/course/)
 * [Intro to Data Structures and Algorithms](https://www.udacity.com/course/data-structures-and-algorithms-in-python--ud513) - Brynn Claypoole, Horatio Thomas (Udacity)
 * [JavaScript Algorithms and Data Structures](https://www.youtube.com/playlist?list=PLC3y8-rFHvwjPxNAKvZpdnsr41E0fCMMP) - Codevolution
+* [JavaScript Algorithms and Data Structures](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures) - FreeCodeCamp
```
Comment (eshellman) [inline courses/free-courses-en.md:140]: duplicate

File: courses/free-courses-en.md [hunk 3]
```diff
@@ -855,6 +857,7 @@
 * [Machine Learning Tutorial in Python](https://youtube.com/playlist?list=PL9ooVrP1hQOHUfd-g8GUpKI3hHOwM_9Dn) - edureka!
 * [Machine Learning Tutorial Python \| Machine Learning For Beginners](https://www.youtube.com/playlist?list=PLeo1K3hjS3uvCeTYTeyfe0-rN5r8zn9rw) - Dhaval Patel
 * [Machine Learning with Python](https://cognitiveclass.ai/courses/machine-learning-with-python) - Saeed Aghabozorgi (cognitiveclass.ai)
+* [Machine Learning with Python](https://www.freecodecamp.org/learn/machine-learning-with-python) - FreeCodeCamp
```
Comment (eshellman) [inline courses/free-courses-en.md:860]: move to interactive tutorials

**Review by eshellman — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md (hunk 2 above)
Comment (MeenakshyBS) [reply courses/free-courses-en.md]: JavaScript Algorithms and Data Structures Certification of FreeCodeCamp wasn't there in the course list before. Kindly check

Comment (MeenakshyBS) [reply courses/free-courses-en.md]: same title issue has been resolved

File: courses/free-courses-en.md (hunk 1 above)
Comment (MeenakshyBS) [reply courses/free-courses-en.md]: same title issue has been resolved.

File: courses/free-courses-en.md (hunk 3 above)
Comment (MeenakshyBS) [reply courses/free-courses-en.md]: Didn't find an interactive tutorial section. I have add interactive tutorial to the title.

Comment (eshellman) [reply courses/free-courses-en.md]: No, it's a separate list.

**Review by eshellman — CHANGES_REQUESTED** [review]
move one resource to interactive tutorial list

**Comment by eshellman** [comment]
resource moved in #9195

**Pull request closed without merging**